### ISSUE TITLE
Cache IPFS results

### ISSUE THREAD

**Issue by wanderingstan** [OriginProtocol/demo-dapp#66]
We're making multiple calls to the server that probably could be cached locally. 

This looks like a good option: 
https://www.npmjs.com/package/object-cache 

Here, IPFS devs are recommending such a solution:
https://github.com/ipfs/js-ipfs-api/issues/511

This could be browser-native package, but it seems to be too tightly connected to the actual web request, that it wouldn't work with ipfs-js:
https://developer.mozilla.org/en-US/docs/Web/API/Cache

 This is the spot in our code where we get from IPFS:
https://github.com/OriginProtocol/demo-dapp/blob/master/src/services/ipfs-service.js#L53

**Comment by DanielVF** [comment]
One additional speed consideration is that we are parsing the listing JSON again every time we render a react listing-card or listing-detail. That might be best done once.

**Comment by wanderingstan** [comment]
I did some timing, and it looks like the IPFS fetching is definitely the culprit. It takes ~3 seconds where every other step is a few milliseconds. 

Full spreadsheet here:
https://docs.google.com/spreadsheets/d/1gi0oUXb3FazJVFTTnSrH2IY4OcMHGp1CrjK9w2X4800/edit?usp=sharing

```
componentWillMount: |  
listing-detail.js:34 loadListing(): | 0.001
listing-detail.js:94 render(): | 0.001
listing-detail.js:89 componentDidMount: | 0.005
listing-detail.js:37 after loadListing().contractService.getListing: | 0.012
listing-detail.js:94 render(): | 0.001
listing-detail.js:41 after loadListing().ipfsService.getListing: | 2.567
listing-detail.js:43 after loadListing().ipfsService.getListing.JSON.parse: | 0.003
listing-detail.js:94 render(): | 0.001
```

**Comment by wanderingstan** [comment]
I tried adding `map-cache` npm module and --wow!-- its fast now.  With a cache hit, we go from 2.567 seconds to 0.005 seconds. 

The fix is in https://github.com/OriginProtocol/demo-dapp/tree/issue/66_cache_ipfs

Can you guys bang on it a bit and see if it broke things? Logic is pretty simple, so I'm too worried.

**Comment by wanderingstan** [comment]
Merged de65f36adce92ddca4c2b5c7a251ec5f28455495 into Master.
